p2p/proto/transaction.proto

```diff
@@ -144,16 +144,10 @@ message TransactionsRequest {
     Iteration iteration = 1;
 }
 
-// can be several in a single reply
-message Transactions {
-    repeated Transaction items = 1;
-}
-
+// Responses are sent ordered by the order given in the request.
 message TransactionsResponse {
-    optional BlockID id = 1; // may not appear if Fin is sent to end the whole response
```

Comment:
AFAIU, `BlockID` is being removed as we do not require delimiting between data of different blocks assuming the block header will be used to determine how many transactions/events belong to which block. However, this relies on trusting the header, and since we cannot calculate the block hash without the transactions, event or state we cannot trust the information contained in the header. Therefore, we cannot use the header to determine when one block's data ends and another's starts.

Hence, IMO, `BlockID` should be added back so that the sender tells the receiver which data belongs to which block. Note, that this doesn't require delimitting between blocks if we assume there is no interleaving.